src/redux/slices/loginSlice.ts

```diff
@@ -0,0 +1,74 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+import { BACKEND_URL } from '../../constants/api';
+
+
+
+interface UserData {
+    email: string;
+    password: string;
+}
+
+interface UserInfoInterface extends UserData {
+    message?: string;
+}
+
+export interface AuthState {
+    userInfo: UserInfoInterface | null;
+    loading: boolean;
+    error: any;
+    isSucceeded: boolean;
+}
+
+export const loginUser = createAsyncThunk<UserData, UserData, { rejectValue: any }>(
+    'login/loginUser',
+    async (userData, { rejectWithValue }) => {
+        try {
+            const response = await axios.post(`${BACKEND_URL}/api/users/login`, userData);
+            localStorage.setItem('token', response.data.token);
```

Comment:
It is advisable to hash the token so everyone in production can't see it. But this is okay for now